### PR TITLE
Speechllm 2312 paarth

### PR DESCRIPTION
**Pull request by paarthneekhara**

### CODE REVIEW

File: nemo/collections/nlp/data/language_modeling/megatron/t5_speechlm_dataset.py [hunk 1]
```diff
@@ -124,6 +124,8 @@ def __init__(
         spec_aug_time_masks = 2,
         cross_attention_epsilon: Optional[float] = 0.0,
         lm_vocab_size: Optional[int] = None,
+        num_speech_codebooks: Optional[int] = 8,
+        codebook_fps: Optional[int] = 76,
```
Comment (blisc) [inline nemo/collections/nlp/data/language_modeling/megatron/t5_speechlm_dataset.py]: We should default this to 75 and just do codebook_fps+1 in the class itself

Comment (paarthneekhara) [reply nemo/collections/nlp/data/language_modeling/megatron/t5_speechlm_dataset.py]: Sure, let me make that change.

Comment (paarthneekhara) [reply nemo/collections/nlp/data/language_modeling/megatron/t5_speechlm_dataset.py]: Updated

File: nemo/collections/nlp/data/language_modeling/megatron/t5_speechlm_dataset.py [hunk 2]
```diff
@@ -286,7 +290,7 @@ def load_data(self, dataset):
 
             if doc["answer_type"] in ["SPEECH", "AUDIOCODEC"]:
                 assert "answer_duration" in doc, f"answer_duration key not in document {doc}"
-                approx_answer_len = doc["answer_duration"] * 76
+                approx_answer_len = doc["answer_duration"] * self.codebook_fps
```
Comment (blisc) [inline nemo/collections/nlp/data/language_modeling/megatron/t5_speechlm_dataset.py]: Let's make this +1 too

File: nemo/collections/nlp/models/language_modeling/megatron_t5_speechlm_model.py [hunk 3]
```diff
@@ -1084,6 +1103,9 @@ def build_virtual_prompt_dataset(
             use_attention_prior=self.cfg.data.get('use_attention_prior', False),
             attention_prior_scaling_factor=self.cfg.data.get('attention_prior_scaling_factor', 1.0),
             cross_attention_epsilon=self.cfg.data.get('cross_attention_epsilon', 0.0),
+            lm_vocab_size=self.lm_vocab_size,
+            num_speech_codebooks=self.num_speech_codebooks,
+            codebook_fps=self.cfg.data.get('codebook_fps', 76),
```
Comment (blisc) [inline nemo/collections/nlp/models/language_modeling/megatron_t5_speechlm_model.py]: Let's default this to 75 too

Comment (paarthneekhara) [reply nemo/collections/nlp/models/language_modeling/megatron_t5_speechlm_model.py]: Thanks for noticing, done.

File: nemo/collections/nlp/data/language_modeling/megatron/t5_speechlm_dataset.py (hunk 2 above)
Comment (paarthneekhara) [reply nemo/collections/nlp/data/language_modeling/megatron/t5_speechlm_dataset.py]: Done.

**Pull request merged**